src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -484,77 +371,165 @@ class HierarchicalGrids : public ParticleContainerInterface<Particle> {
+      collectParticles.insert(collectParticles.end(), collectParticlesOfLevel.begin(), collectParticlesOfLevel.end());
+    }
 
-    return _hierarchyLevels[0].updateContainer(keepNeighborListsValid);
-
+    return collectParticles;
   }
 
   /**
    * @copydoc autopas::ParticleContainerInterface::getTraversalSelectorInfo()
    */
   [[nodiscard]] TraversalSelectorInfo getTraversalSelectorInfo() const override {
-
     return _hierarchyLevels[0].getTraversalSelectorInfo();
-
   }
 
   /**
    * @copydoc autopas::ParticleContainerInterface::getParticle()
    */
   std::tuple<const Particle *, size_t, size_t> getParticle(size_t cellIndex, size_t particleIndex,
-                                                            IteratorBehavior iteratorBehavior) const override {
-
-    return _hierarchyLevels[0].getParticle(cellIndex, particleIndex, iteratorBehavior); 
-
+                                                           IteratorBehavior iteratorBehavior,
+                                                           const std::array<double, 3> &boxMin,
+                                                           const std::array<double, 3> &boxMax) const override {
+    return getParticleImpl<true>(cellIndex, particleIndex, iteratorBehavior, boxMin, boxMax);
   }
 
   /**
    * @copydoc autopas::ParticleContainerInterface::getParticle()
    */
   std::tuple<const Particle *, size_t, size_t> getParticle(size_t cellIndex, size_t particleIndex,
-                                                                   IteratorBehavior iteratorBehavior,
-                                                                   const std::array<double, 3> &boxMin,
-                                                                   const std::array<double, 3> &boxMax) const override {
+                                                           IteratorBehavior iteratorBehavior) const override {
+    // this is not a region iter hence we stretch the bounding box to the numeric max
+    constexpr std::array<double, 3> boxMin{std::numeric_limits<double>::lowest(), std::numeric_limits<double>::lowest(),
+                                           std::numeric_limits<double>::lowest()};
+
+    constexpr std::array<double, 3> boxMax{std::numeric_limits<double>::max(), std::numeric_limits<double>::max(),
+                                           std::numeric_limits<double>::max()};
+    return getParticleImpl<false>(cellIndex, particleIndex, iteratorBehavior, boxMin, boxMax);
+  }
+
+  /**
+   * Container specific implementation for getParticle. See ParticleContainerInterface::getParticle().
+   *
+   * @tparam regionIter
+   * @param cellIndex
+   * @param particleIndex
+   * @param iteratorBehavior
+   * @param boxMin
+   * @param boxMax
+   * @return tuple<ParticlePointer, CellIndex, ParticleIndex>
+   */
+  template <bool regionIter>
+  std::tuple<const Particle *, size_t, size_t> getParticleImpl(size_t encodedLevelAndCellIndex, size_t particleIndex,
```

Comment:
I'd like to see a dedicated unit test that shows that this works over several levels